packages/engine/src/assets/loaders/gltf/extensions/EEECSImporterExtension.ts

```diff
@@ -17,8 +21,16 @@ export default class EEECSImporterExtension extends ImporterExtension implements
     const nodeDef = json.nodes[nodeIndex]
     if (!nodeDef.extensions?.[this.name]) return null
     const extensionDef: EE_ecs = nodeDef.extensions[this.name]
-    if (!parser.options.entity) throw Error('error loading ecs data')
-    parseECSData(parser.options.entity, Object.entries(extensionDef.data))
+    const entityName = extensionDef.data?.['xrengine.entity'] as string
+    if (!entityName) return null
+    let entity: Entity
+    if (NameComponent.entitiesByName[entityName]) {
```

Comment:
Names are not guaranteed to be unique

Reply:
entitiesByName is a record object, so names are guaranteed to be unique since only one entity can be mapped to any name. If that is not the intended behavior then the NameComponent needs changes

Reply:
Names are there as a convenience, primarily for debugging purposes, and do not guarantee uniqueness. If a second entity is added with the same name, it will override the previous one in the record.

Reply:
UUIDComponent provides a unique id

Reply:
That should not be the behavior in any case. I've modified the entitiesByName object to map names to an array of objects, which allows multiple entities with the same name without loss of information. I've then removed the above check since multiple entities with the same name is no longer undefined behavior. There is a corresponding PR in the puttclub project that adjusts the code for this change